compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -1623,6 +1654,110 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
         }
     }
 
+    /// Given an attribute macro that failed to be resolved, look for `derive` macros that could
+    /// provide it, either as-is or with small typos.
+    fn detect_derive_attribute(
+        &self,
+        err: &mut Diag<'_>,
+        ident: Ident,
+        parent_scope: &ParentScope<'ra>,
+        sugg_span: Option<Span>,
+    ) {
+        // Find all of the `derive`s in scope and collect their corresponding declared
+        // attributes.
+        // FIXME: this only works if the crate that owns the macro that has the helper_attr
+        // has already been imported.
+        let mut derives = vec![];
+        let mut all_attrs: FxHashMap<Symbol, Vec<_>> = FxHashMap::default();
```

Comment:
Tip: You could use `UnordMap` here to make it clear that you will never iterate over this list. `UnordMap` is literally just a wrapper around `FxHashMap` that prevents you from iterating over it.